src/Umbraco.Web.UI.Client/src/views/prevalueeditors/treesource.html

```diff
@@ -85,77 +86,51 @@ <h5><localize key="contentPicker_configurationXPathTitle">XPath Query</localize>
-                {{ ("umb://" + (model.value.type === 'content' ? 'document' : model.value.type) + "/" + model.value.dynamicRoot.originKey | ncNodeName)}}
-              </div>
-            </div>
-          </div>
-          <div class="umb-node-preview__actions">
-            <button type="button" class="umb-node-preview__action" ng-click="openDynamicRootOriginPicker()"><localize key="general_change">Change</localize></button>
-          </div>
-        </div>
-
-        <!-- list of query steps -->
-        <div ui-sortable="sortableOptionsForQuerySteps" ng-model="model.value.dynamicRoot.querySteps">
-          <div class="umb-node-preview" single ng-repeat="queryStep in model.value.dynamicRoot.querySteps track by $index">
-            <div class="flex">
-              <umb-icon class="umb-node-preview__icon" icon="{{getIconForQueryStepAlias(queryStep.alias)}}"></umb-icon>
-              <div class="umb-node-preview__content">
-                <div class="umb-node-preview__name">
-                  <localize ng-if="queryStep.alias === 'NearestAncestorOrSelf' ||
-                  queryStep.alias === 'FurthestAncestorOrSelf' ||
-                  queryStep.alias === 'NearestDescendantOrSelf' ||
-                  queryStep.alias === 'FurthestDescendantOrSelf'" key="dynamicRoot_queryStep{{queryStep.alias}}Title"></localize>
-                  <span ng-if="queryStep.alias !== 'NearestAncestorOrSelf' &&
-                  queryStep.alias !== 'FurthestAncestorOrSelf' &&
-                  queryStep.alias !== 'NearestDescendantOrSelf' &&
-                  queryStep.alias !== 'FurthestDescendantOrSelf'">{{queryStep.alias}}</span>
-                </div>
-                <div class="umb-node-preview__description" ng-if="queryStep.anyOfDocTypeKeys && queryStep.anyOfDocTypeKeys.length > 0">
-                  <!-- todo: Maybe get the name to display here. -->
-                  <localize key="dynamicRoot_queryStepTypes">of type: </localize>
-
-                  <span ng-repeat="key in queryStep.anyOfDocTypeKeys track by $index">
-                    {{ key | umbCmsJoinArray:', '}}
-                  </span>
-                </div>
-              </div>
-            </div>
-            <div class="umb-node-preview__actions">
-              <button type="button" class="umb-node-preview__action" ng-click="removeQueryStep(queryStep)"><localize key="general_remove">Remove</localize></button>
-            </div>
-          </div>
+        <!-- Origin -->
+        <umb-node-preview
+            single
+            icon="vm.dynamicRootOrigin.icon"
+            name="vm.dynamicRootOrigin.name"
+            description="vm.dynamicRootOrigin.description"
+            allow-change="true"
+            allow-remove="false"
+            on-change="vm.openDynamicRootOriginPicker()">
+        </umb-node-preview>
+
+        <!-- List of query steps -->
+        <div ui-sortable="vm.sortableOptionsForQuerySteps" ng-model="vm.sortableModel" ng-if="vm.querySteps">
```

Comment:
`ui-sortable` require `ng-model` and it use a separate model like in Content Picker / MNTP.